koa_test.go

```diff
@@ -0,0 +1,372 @@
+					},
+					{
+						RawByte: []byte{0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x08},
+						Value:   "0000000000000008",
+					},
+					{
+						RawByte: []byte{byte(opcode.Push)},
+						Value:   "Push",
+					},
+					{
+						RawByte: []byte{0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00},
+						Value:   "0000000000000000",
+					},
+					{
+						RawByte: []byte{byte(opcode.Mload)},
+						Value:   "Mload",
+					},
+					{
+						RawByte: []byte{byte(opcode.Push)},
+						Value:   "Push",
+					},
+					{
+						RawByte: []byte{0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x08},
+						Value:   "0000000000000008",
+					},
+					{
+						RawByte: []byte{byte(opcode.Push)},
+						Value:   "Push",
+					},
+					{
+						RawByte: []byte{0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x08},
+						Value:   "0000000000000008",
+					},
+					{
+						RawByte: []byte{byte(opcode.Mload)},
+						Value:   "Mload",
+					},
+					{
+						RawByte: []byte{byte(opcode.Push)},
+						Value:   "Push",
+					},
+					{
+						RawByte: []byte{0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x08},
+						Value:   "0000000000000008",
+					},
+					{
+						RawByte: []byte{byte(opcode.Push)},
+						Value:   "Push",
+					},
+					{
+						RawByte: []byte{0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x10},
+						Value:   "0000000000000010",
+					},
+					{
+						RawByte: []byte{byte(opcode.Mload)},
+						Value:   "Mload",
+					},
+					{
+						RawByte: []byte{byte(opcode.Sub)},
+						Value:   "Sub",
+					},
+					{
+						RawByte: []byte{byte(opcode.Push)},
+						Value:   "Push",
+					},
+					{
+						RawByte: []byte{0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x08},
+						Value:   "0000000000000008",
+					},
+					{
+						RawByte: []byte{byte(opcode.Push)},
+						Value:   "Push",
+					},
+					{
+						RawByte: []byte{0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x18},
+						Value:   "0000000000000018",
+					},
+					{
+						RawByte: []byte{byte(opcode.Mload)},
+						Value:   "Mload",
+					},
+					{
+						RawByte: []byte{byte(opcode.Mul)},
+						Value:   "Mul",
+					},
+					{
+						RawByte: []byte{byte(opcode.Add)},
+						Value:   "Add",
+					},
+					{
+						RawByte: []byte{byte(opcode.Returning)},
+						Value:   "Returning",
+					},
+				},
+			},
+			err: nil,
+		},
+	}
+}
+
+func readFile(fileName string) (string, error) {
+	file, err := os.OpenFile(fileName, os.O_RDONLY, os.FileMode(644))
+	if err != nil {
+		return "", err
+	}
+
+	fs, err := file.Stat()
+	if err != nil {
+		return "", err
+	}
+
+	data := make([]byte, fs.Size())
+	if err != nil {
+		return "", err
+	}
+
+	_, err = file.Read(data)
+	if err != nil {
+		return "", err
+	}
+
+	return string(data), nil
+}
+
+func TestCompile(t *testing.T) {
+	tests := defineAsm()
+
+	for i, test := range tests {
+		str, err := readFile(test.fileName)
+		if err != nil {
+			continue
+		}
+
+		asm, err := Compile(str)
+
+		// TODO : after implements all functions in compile
```

Comment:
What is todo?

Reply:
This is just TODO..because we currently are not implementing compile function and call.
So I want to mark TODO for making other cases after making those functions.

Reply:
ok!